step/step.go

```diff
@@ -0,0 +1,566 @@
+	IsSingleBuild         bool   `env:"single_build,opt[true,false]"`
+	ShouldBuildBeforeTest bool   `env:"should_build_before_test,opt[yes,no]"`
+
+	RetryTestsOnFailure       bool `env:"should_retry_test_on_fail,opt[yes,no]"`
+	DisableIndexWhileBuilding bool `env:"disable_index_while_building,opt[yes,no]"`
+	GenerateCodeCoverageFiles bool `env:"generate_code_coverage_files,opt[yes,no]"`
+	HeadlessMode              bool `env:"headless_mode,opt[yes,no]"`
+
+	TestOptions         string `env:"xcodebuild_test_options"`
+	XcprettyTestOptions string `env:"xcpretty_test_options"`
+
+	// Debug
+	Verbose                     bool   `env:"verbose,opt[yes,no]"`
+	CollectSimulatorDiagnostics string `env:"collect_simulator_diagnostics,opt[always,on_failure,never]"`
+
+	// Output export
+	DeployDir             string `env:"BITRISE_DEPLOY_DIR"`
+	ExportUITestArtifacts bool   `env:"export_uitest_artifacts,opt[true,false]"`
+
+	CacheLevel string `env:"cache_level,opt[none,swift_packages]"`
+}
+
+type exportCondition string
+
+const (
+	always    = "always"
+	never     = "never"
+	onFailure = "on_failure"
+)
+
+// Config ...
+type Config struct {
+	ProjectPath string
+	Scheme      string
+	TestPlan    string
+
+	XcodeMajorVersion int
+	SimulatorID       string
+	IsSimulatorBooted bool
+
+	TestRepetitionMode            string
+	MaximumTestRepetitions        int
+	RelaunchTestForEachRepetition bool
+
+	OutputTool         string
+	IsCleanBuild       bool
+	IsSingleBuild      bool
+	BuildBeforeTesting bool
+
+	RetryTestsOnFailure       bool
+	DisableIndexWhileBuilding bool
+	GenerateCodeCoverageFiles bool
+	HeadlessMode              bool
+
+	XcodebuildTestOptions string
+	XcprettyOptions       string
+
+	SimulatorDebug exportCondition
+
+	DeployDir             string
+	ExportUITestArtifacts bool
+
+	CacheLevel string
+}
+
+// XcodeTestRunner ...
+type XcodeTestRunner struct {
+	inputParser       stepconf.InputParser
+	logger            log.Logger
+	xcprettyInstaller xcpretty.Installer
+	xcodebuild        xcodebuild.Xcodebuild
+	simulatorManager  simulator.Manager
+	cache             cache.SwiftPackageCache
+	outputExporter    output.Exporter
+	pathModifier      pathutil.PathModifier
+}
+
+// NewXcodeTestRunner ...
+func NewXcodeTestRunner(inputParser stepconf.InputParser, logger log.Logger, xcprettyInstaller xcpretty.Installer, xcodebuild xcodebuild.Xcodebuild, simulatorManager simulator.Manager, cache cache.SwiftPackageCache, outputExporter output.Exporter, pathModifier pathutil.PathModifier) XcodeTestRunner {
+	return XcodeTestRunner{
+		inputParser:       inputParser,
+		logger:            logger,
+		xcprettyInstaller: xcprettyInstaller,
+		xcodebuild:        xcodebuild,
+		simulatorManager:  simulatorManager,
+		cache:             cache,
+		outputExporter:    outputExporter,
+		pathModifier:      pathModifier,
+	}
+}
+
+func (s XcodeTestRunner) validateXcodeVersion(input *Input, xcodeMajorVersion int) error {
```

Comment:
Not sure wdyt, but I'd organize the methods like 
```
Consts, Types, Structs whatsoever

// Main methods
ProcessConfig
InstallDeps
Run
Export

// Helper methods for Process (if any)
validateXYZ

// Helper methods for Install (if any)

// Helper methods for Run (if any)

// Helper methods for Export (if any)
```

For me this is the logical order of "importance"

Reply:
As discussed we will get back to this topic later.